### DISCUSSION TITLE
Is it possible to filter items in data hub dialog?

### DISCUSSION THREAD

**Discussion by q-nicolechoi** [microsoft/Microsoft-Fabric-workload-development-sample#157] [Help]
I want to filter the table of items in the data hub dialog. The [docs](https://learn.microsoft.com/en-us/javascript/api/@ms-fabric/workload-client/datahubapi?view=fabric-extensibility-js-latest#@ms-fabric-workload-client-datahubapi-opendialog) suggest that I can use `shouldIncludeFuncByItemType` in the dialog, e.g.
```
workloadClient.client.datahub.openDialog({
    ...,
    supportedTypes: ['Lakehouse'],
    shouldIncludeFuncByItemType: {
        Lakehouse: (item) => {
            return true; // e.g. item.displayName.includes('abc') || !item.properties.defaultSchema
        }
    },
})
```
But I got an error in the console but I wasn't sure what really causes the error, or if this is the correct way at all to filter the table in data hub dialog. Has anyone been able to get this to work before?
```
ERROR DataCloneError: Failed to execute 'postMessage' on 'Window': item => {
          return true;
        } could not be cloned.
    at MessageBroker.sendMessage (workload-client.js:783:46)
    at workload-client.js:734:26
    at new ZoneAwarePromise (zone.js:1411:21)
    at MessageBroker.<anonymous> (workload-client.js:728:24)
    at Generator.next (<anonymous>)
    at tslib.es6.js:121:1
    at new ZoneAwarePromise (zone.js:1411:21)
    at Module.__awaiter (tslib.es6.js:117:1)
    at MessageBroker.sendAsync (workload-client.js:727:26)
    at WorkloadClientMessageBrokerImpl.<anonymous> (workload-client.js:829:36)
```

**Comment by vadimazarov** [comment]
Hi Nicole,
The sample code has an example of filtering by type (where you select either Lakehouse or the SampleItem - https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/blob/c59c0db7868c80ec879e031d6d44412e0413f462/Frontend/src/controller/SampleWorkloadController.ts#L295

Could you check if it works for you with the sample code, out of the box? Or try without the 'shouldIncludeFuncByItemType' value.
Do reach out if the issue persists.
Thanks!
Vadim

**Reply by q-nicolechoi** [reply]
Hi Vadim, originally we use the following and it works. It is when I try to filter the lakehouses using `shouldIncludeFuncByItemType` that gave me an error
```
const datahubConfig: DatahubSelectorDialogConfig = {
            supportedTypes: ['Lakehouse'],
            multiSelectionEnabled: false,
            dialogDescription: 'description',
            dialogAdditionalInfo: 'additional info',
            hostDetails: {
                experience: 'experience',
                scenario: 'scenario',
            },
            learnMorelink: {
                linkUrl: 'https://learn.microsoft.com/en-us/fabric/get-started/onelake-data-hub',
                linkText: 'Learn more about OneLake data hub'
            }
        };
```

**Reply by IdanGoldfarb1989** [reply]
Hi Nicole,

I can reproduce the issue on my side, and it appears to be a bug.

For now, this functionality is not supported. We will either fix it or remove it from the SDK.

Thanks,
Idan

**Reply by q-nicolechoi** [reply]
Hi Idan,

It will be useful to know if this is supported in the long run, so once a decision has been made, would you mind coming back to this thread and give us an update please :)

Thank you,
Nicole
